pkg/agent/proxy/proxier.go

```diff
@@ -618,6 +619,11 @@ func (p *proxier) installServices() {
 				}
 				// Install LoadBalancer flows and configurations.
 				if len(toAdd) > 0 {
+					groupID, exists = p.groupCounter.Get(svcPortName, externalPolicyLocal)
```

Comment:
not introduced by this PR, but I see a lot of call to `p.groupCounter.Get` to check for group existence. Aren't some of these calls redundant? Why not call the function once (for each of the 2 groups), or use the result of `AllocateIfNotExist`?

Reply:
I feel the same. I'm planning a separate patch to reduce redundancy in this method, will push for review soon.